code/datums/actions/items/unzip.dm

```diff
@@ -0,0 +1,16 @@
+/datum/action/item_action/zipper
+	name = "Un-Zip Duffel"
+	desc = "Un-Zip the equipped duffelbag so you can root around in it"
+
+/datum/action/item_action/zipper/New(Target)
+	. = ..()
+	RegisterSignal(target, COMSIG_DUFFEL_ZIP_CHANGE, PROC_REF(on_zip_change))
+	var/obj/item/storage/backpack/duffelbag/duffle_target = target
+	on_zip_change(target, duffle_target.zipped_up)
+
+/datum/action/item_action/zipper/proc/on_zip_change(datum/source, new_zip)
+	SIGNAL_HANDLER
+	var/zip = new_zip ? "Un-Zip" : "Zip"
+	name = "[zip] Duffel"
+	desc = "[zip] the equipped duffelbag so you can root around in it"
```

Comment:
```suggestion
	if(new_zip)
		name = "Unzip" 
		desc = "Unzip your equipped duffelbag so you can access its contents."
	else
		name = "Zip"
		desc = "Zip your equipped duffelbag so you can move around faster."
```
Desc probably shouldn't read out "so you can root around in it" for when the action is "zipping"

Reply:
ah fuck forgot about that